bot/seasons/easter/egg_decorating.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import random
+from io import BytesIO
+from pathlib import Path
+
+
+import discord
+from PIL import Image
+from discord.ext import commands
+
+log = logging.getLogger(__name__)
+
+COLOURS = [
+    (255, 0, 0, 255), (255, 128, 0, 255), (255, 255, 0, 255), (0, 255, 0, 255),
+    (0, 255, 255, 255), (0, 0, 255, 255), (255, 0, 255, 255), (128, 0, 128, 255)
+]  # Colours to be replaced - Red, Orange, Yellow, Green, Light Blue, Dark Blue, Pink, Purple
+
+IRREPLACEABLE = [
+    (0, 0, 0, 0), (0, 0, 0, 255)
+]  # Colours that are meant to stay the same - Transparent and Black
+
+
+class EggDecorating(commands.Cog):
+    """A Command that lets you decorate some easter eggs!"""
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(aliases=["decorateegg"])
+    async def eggdecorate(self, ctx, *colours):
+        """
+        This 'paints' a beautiful egg using inputted colours
+
+        It picks from a random set of designs and alters the colours to the user's liking
+        """
+
+        if len(colours) < 2:
+            return await ctx.send("You must include at least 2 colours!")
+
+        invalid = []
+        converted = []
+        for c in colours:
+            try:
+                colour = await commands.ColourConverter().convert(ctx, c)
+                # Attempts to convert the arguments into discord.Colour
+                converted.append(colour)
+            except commands.BadArgument:
+                invalid.append(c)
```

Comment:
If we added the type hinted converter properly, we don't need to manually convert to discord.Colour for those that are supported. 

For anything that isn't, we can put the colour through the recommended helper method from before that tries to match the colour with the HTML or XKCD colour names. It will return None if there's no match, so it gets appended to `invalid` like before, otherwise it just replaces the current item in the list by index.

```py
        invalid = []
        colours = list(colours)
        for idx, colour in enumerate(colours):
            if isinstance(colour, discord.Colour):
                continue
            value = self.replace_invalid(colour)
            if value:
                colours[idx] = discord.Colour(value)
            else:
                invalid.append(colour)
```